json/src/test/scala/spray/json/BasicFormatsSpec.scala

```diff
@@ -30,42 +30,18 @@ class BasicFormatsSpec extends WordSpec with DefaultJsonProtocol {
     "convert a Float to a JsNumber" in {
       4.2f.toJson shouldEqual JsNumber(4.2f)
     }
-    "convert a Float.NaN to a JsNull" in {
-      Float.NaN.toJson shouldEqual JsNull
-    }
-    "convert a Float.PositiveInfinity to a JsNull" in {
-      Float.PositiveInfinity.toJson shouldEqual JsNull
-    }
-    "convert a Float.NegativeInfinity to a JsNull" in {
-      Float.NegativeInfinity.toJson shouldEqual JsNull
-    }
     "convert a JsNumber to a Float" in {
       JsNumber(4.2f).convertTo[Float] shouldEqual 4.2f
     }
-    "convert a JsNull to a Float" in {
-      JsNull.convertTo[Float].isNaN shouldEqual Float.NaN.isNaN
-    }
   }
 
   "The DoubleJsonFormat" should {
     "convert a Double to a JsNumber" in {
       4.2.toJson shouldEqual JsNumber(4.2)
     }
-    "convert a Double.NaN to a JsNull" in {
```

Comment:
Why did you remove these tests?

Reply:
We no longer support NaN since we go via BigDecimal.